### ISSUE TITLE
Support for Title Case

### ISSUE THREAD

**Issue by mimmiechan** [pietervanheijningen/clickbait-remover-for-youtube#43]
The "capitalize words" option does not cut it at all for me. I need proper [title case](https://en.wikipedia.org/wiki/Title_case). That is where certain small words, like "a", "the" and "for" are written in only lower-case. This is unfortunately not possible with CSS. I made a userscript to fix this issue. Well, two scripts actually, trying different code. Feel free to implement it.
I haven't learned javascript properly, so this might be awful code. And it only works for channel pages right now.

[Title Case for YouTube videos.js](https://gist.github.com/mimmiechan/0ae9f53fd68a4cd9eeb0c5df30ad1724)
[Title Case for YouTube videos (ap-style-title-case).js](https://gist.github.com/mimmiechan/f4e8bddfc6d94d8cc4bd951a2519b015)

There is an option in the code to allow all caps words. The main advantage of that, I think, is that abbreviations like "FBI" or "FPS" (or whatever) will be properly cased.

Thanks for the addon!

**Comment by pietervanheijningen** [comment]
Thanks for putting in the effort! Think i'll refrain from implementing it though, as to me getting most of the way there with just CSS is much cleaner. Especially as youtube has a lot of video suggestions that are generated as you scroll down. If you want to change those titles with javascript you would have a loop on an interval which would be expensive and cause "popping" by titles changing.

**Comment by muhdiboy** [comment]
Maybe not exactly "Title Case", but is it possible to add a feature to lowercase words but leave the first letter of capitilized words still capitilized.
This would result in following behaviour:
* FOO BAR -> Foo Bar
* FOO Bar -> Foo Bar
* Foo bar -> Foo bar
* foo bar -> foo bar

**Comment by alpe12** [comment]
> If you want to change those titles with javascript you would have a loop on an interval which would be expensive and cause "popping" by titles changing.

Not if using MutationObserver.

**Comment by pietervanheijningen** [comment]
> > If you want to change those titles with javascript you would have a loop on an interval which would be expensive and cause "popping" by titles changing.
> 
> Not if using MutationObserver.

Yeah you're actually right. Might look into it if i get around to it.
